HW1/P4/P4_bfs.py

```diff
@@ -0,0 +1,69 @@
+# this part goes in P4_bfs
+# make function taking in graph RDD and char name
+# gives shortest path to all other nodes
+# diameter <=10
+
+
+# initiallizing spark
+from pyspark import SparkContext, SparkConf
+sc = SparkContext(conf=SparkConf())
+sc.setLogLevel("ERROR")
+
+
+
+# outside calculations because of lazy RDD updating
+def updatedist(children,visited,count):
+    print count
+    children = [j for j in children if j not in visited]
+    return lambda x: (x[0],count) if x[0][1] in children else x
+
+
+# x is [visited,parents,children]
+def updatefamily(x,graphRDD):
+    x[2] = [j for j in x[2] if j not in x[0]]
+    x[0] += x[2][:]
+    x[1] = x[2][:]
+    x[2] = []
```

Comment:
These operations are executed in python, but should be executed by spark's functions instead.